### ISSUE TITLE
Support for SSG/SSR?

### ISSUE THREAD

**Issue by Blackglade** [utkarshdubey/react-spline#14]
Using React-based frameworks like NextJS or GatsbyJS causes issues with server side rendering / generation due to the fact that the window obj is undefined at build time (see attached screenshot). Possible to patch a fix that prevents this issue?
![Screen Shot 2021-08-08 at 5 09 48 PM](https://user-images.githubusercontent.com/6363077/128649851-ba4b26a4-0f1a-41ff-a671-c6d5dfcbeb47.png)

**Comment by utkarshdubey** [comment]
Not really, but this looks interesting. We'd have to rewrite the runtime in a way that doesn't rely on window variables.

**Comment by adalbertopc** [comment]
You can use `react-spline` on NextJs by using this:

`import dynamic from "next/dynamic";`

`const DynamicComponentWithNoSSR = dynamic(
  () => import("../components/spline/MyComponent"),
  { ssr: false }
);
`

Then use it as a normal component

`<DynamicComponentWithNoSSR />`

**Comment by Blackglade** [comment]
Thanks @adalbertopc. I ended up doing something similar to this.
